### PR TITLE
 [dev] AB#33495 - New Select Component

### PR DESCRIPTION
**Pull request by morethanfire**
New Select component that was pulled from Dropdown. There isn't any unit tests and documentation for this yet, because it was to aid in helping me out with my Surfaces -> FiltersRail component. I figured it wouldn't hurt to introduce into 9.17.0 though. The plan would be to create documentation and do unit tests in version 10.0.0.

### CODE REVIEW

**Comment by groberts314** [comment]
@morethanfire is 9.17.0 going to go into HC 1.29.0, or (hopefully) 1.30.0.
I see that this particular thing should be an innocuous extra thing to help us in the future, but I'm also curious about #207 and [Backlog Item 35112: [UI Component] Radio component should emit the given id into the actual rendered HTML](https://dev.azure.com/saddlebackchurch/Church%20Management/_workitems/edit/35112) which doesn't currently have a Proposed Release Version.

File: src/inputs/select/select.jsx [hunk 1]
```diff
@@ -0,0 +1,831 @@
...
+        valueComponent,
+    } = props;
+
+    const classes = useStyles(props);
+
+    const dropdownMenuRef = useRef();
+    const selectRef = useRef();
+
+    const onChange = (selectedOption) => {
+        console.log('onChangeProp', onChangeProp);
```
Comment (groberts314) [inline src/inputs/select/select.jsx:671]: Let's remove this por favor

File: src/inputs/select/select.jsx [hunk 2]
```diff
@@ -0,0 +1,831 @@
...
+        if (isFunction(onChangeProp)) {
+            console.log('selectedOption', selectedOption);
```
Comment (groberts314) [inline src/inputs/select/select.jsx:673]: And this

File: src/inputs/select/select.jsx [hunk 3]
```diff
@@ -0,0 +1,831 @@
...
+        classes.root,
+        className,
+        {
+            [classes.isFluid]: isFluid,
+            [classes.isUnderlined]: isUnderlined,
+        },
+    );
+    const showRequiredIndicator = required && isEmpty(value);
+
+    const ReactSelectComponent = isCreatable ? ReactSelect.Creatable : CustomReactSelect;
```
Comment (groberts314) [inline src/inputs/select/select.jsx:737]: I'd rather see a blank line between 734 and 735, and think there's less call for one between 735 and 737, but to each his own.  _Not a Blocker_

**Review by groberts314 — CHANGES_REQUESTED** [review]
Overall, looks pretty good.  Just a couple of lingering `console.log()`s have got to go.

Question: Is this intended to replace `<Dropdown selection>`?

File: src/inputs/select/select.jsx (hunk 2 above)
Comment (morethanfire) [reply src/inputs/select/select.jsx]: Fixed.

File: src/inputs/select/select.jsx (hunk 1 above)
Comment (morethanfire) [reply src/inputs/select/select.jsx]: Fixed.

File: src/inputs/select/select.jsx (hunk 3 above)
Comment (morethanfire) [reply src/inputs/select/select.jsx]: Fixed.

**Comment by morethanfire** [comment]
>@morethanfire is 9.17.0 going to go into HC 1.29.0, or (hopefully) 1.30.0.
I see that this particular thing should be an innocuous extra thing to help us in the future, but I'm also curious about #207 and Backlog Item 35112: [UI Component] Radio component should emit the given id into the actual rendered HTML which doesn't currently have a Proposed Release Version.

Finding out from @sircris whether this version, 9.17.0, needs to be in the releave, 1.29.0. I honestly rather it not be and for it to maybe go to 1.30.0. If it does I may need to pull back of the changes.

**Comment by groberts314** [comment]
@morethanfire Could we take just #207 into a 9.17.0 for 1.29.0 (if @sircris were to insist) and hold this one until 9.18.0 and 1.30.0?

File: src/inputs/select/select.jsx (hunk 1 above)
Comment (groberts314) [reply src/inputs/select/select.jsx]: Looks good.  Resolving.

**Comment by morethanfire** [comment]
From Teams @sircris say's that it does not have to be in 1.29.0 release. I will update the PBI mentioned above.

File: src/inputs/select/select.jsx (hunk 2 above)
Comment (groberts314) [reply src/inputs/select/select.jsx]: Looks good.  Resolving.

File: src/inputs/select/select.jsx (hunk 3 above)
Comment (groberts314) [reply src/inputs/select/select.jsx]: Thanks for indulging my whitespace preferences! 😁 
Resolving.

**Review by groberts314 — APPROVED** [review]

**Pull request merged**